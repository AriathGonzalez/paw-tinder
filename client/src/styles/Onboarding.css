.onboarding {
  border-top: solid 0.1em rgba(244, 65, 108, 1);
}

form {
  display: flex;
  justify-content: center;
}

form section {
  display: flex;
  flex-direction: column;
  padding: 1.25em;
  text-align: start;
  width: 35%;
}

form input {
  border: solid 0.1em rgba(244, 65, 108, 1);
  border-radius: 0.6em;
  font-size: 1rem;
  margin: 0.6em 0;
  padding: 1em 1.9em;
}

form input::-webkit-outer-spin-button,
form input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

form label {
  margin: 0.6em 0;
}

form .multiple-input-container {
  display: flex;
  flex-direction: row;
}

form .multiple-input-container input {
  margin-right: 0.6em;
}

form .multiple-input-container label {
  border: solid 0.1em rgba(0, 184, 169, 1);
  border-radius: 0.6em;
  margin-right: 0.6em;
  padding: 0.6em;
  transition: all 0.3s;
}

form input[type="radio"] {
  display: none;
}

form input[type="number"] {
  width: 10%;
}

form input[type="radio"]:checked + label {
  border: solid 0.1em rgba(0, 184, 169, 0.5);
}

form input[type="submit"] {
  cursor: pointer;
}

form input[type="submit"]:hover {
  background-color: rgba(244, 65, 108, 0.75);
}

form input[type="submit"]:active {
  background-color: rgba(244, 65, 108, 1);
}

input:invalid {
  border: solid 0.1em rgba(244, 65, 108, 1);
}

form .photo-container img {
  width: 100%;
}
