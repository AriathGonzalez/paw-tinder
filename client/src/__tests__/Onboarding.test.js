import Onboarding from "../pages/Onboarding";
import renderer from "react-test-renderer";

describe("Onboarding Page", () => {
  describe("Snapshot Test", () => {
    it("Should render correctly", () => {
      const tree = renderer.create(<Onboarding />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
