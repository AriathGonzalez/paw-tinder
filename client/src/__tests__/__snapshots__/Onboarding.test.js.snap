// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Page Snapshot Test Should render correctly 1`] = `
Array [
  <nav>
    <div
      className="logo-container"
    >
      <img
        alt="Logo"
        className="logo"
        src="paw_tinder_logo.png"
      />
    </div>
  </nav>,
  <div
    className="onboarding"
  >
    <h2>
      Create Account
    </h2>
    <form
      onSubmit={[Function]}
    >
      <section>
        <label
          htmlFor="first_name"
        >
          First Name
        </label>
        <input
          id="first_name"
          name="first_name"
          onChange={[Function]}
          placeholder="First Name"
          required={true}
          type="text"
          value=""
        />
        <label>
          Birthday
        </label>
        <div
          className="multiple-input-container"
        >
          <input
            id="dob_day"
            max="31"
            min="1"
            name="dob_day"
            onChange={[Function]}
            placeholder="DD"
            required={true}
            type="number"
            value=""
          />
          <input
            id="dob_month"
            max="12"
            min="1"
            name="dob_month"
            onChange={[Function]}
            placeholder="MM"
            required={true}
            type="number"
            value=""
          />
          <input
            id="dob_year"
            name="dob_year"
            onChange={[Function]}
            placeholder="YYYY"
            required={true}
            type="number"
            value=""
          />
        </div>
        <label>
          Gender
        </label>
        <div
          className="multiple-input-container"
        >
          <input
            checked={true}
            id="man_gender_identity"
            name="gender_identity"
            onChange={[Function]}
            type="radio"
            value="man"
          />
          <label
            htmlFor="man_gender_identity"
          >
            Good Boy
          </label>
          <input
            checked={false}
            id="woman_gender_identity"
            name="gender_identity"
            onChange={[Function]}
            type="radio"
            value="woman"
          />
          <label
            htmlFor="woman_gender_identity"
          >
            Sweet Girl
          </label>
          <input
            checked={false}
            id="more_gender_identity"
            name="gender_identity"
            onChange={[Function]}
            type="radio"
            value="more"
          />
          <label
            htmlFor="more_gender_identity"
          >
            More
          </label>
        </div>
        <label
          htmlFor="show_gender"
        >
          Show gender on my profile
        </label>
        <input
          checked={false}
          id="show_gender"
          name="show_gender"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Show Me
        </label>
        <div
          className="multiple-input-container"
        >
          <input
            checked={false}
            id="man_gender_interest"
            name="gender_interest"
            onChange={[Function]}
            type="radio"
            value="man"
          />
          <label
            htmlFor="man_gender_interest"
          >
            Good Boy
          </label>
          <input
            checked={true}
            id="woman_gender_interest"
            name="gender_interest"
            onChange={[Function]}
            type="radio"
            value="woman"
          />
          <label
            htmlFor="woman_gender_interest"
          >
            Sweet Girl
          </label>
          <input
            checked={false}
            id="everyone_gender_interest"
            name="gender_interest"
            onChange={[Function]}
            type="radio"
            value="everyone"
          />
          <label
            htmlFor="everyone_gender_interest"
          >
            Everyone
          </label>
        </div>
        <label
          htmlFor="about"
        >
          About Me
        </label>
        <input
          id="about"
          name="about"
          onChange={[Function]}
          placeholder="I love playing fetch..."
          required={true}
          type="text"
          value=""
        />
        <input
          type="submit"
        />
      </section>
      <section>
        <label
          htmlFor="about"
        >
          Profile
        </label>
        <input
          id="url"
          name="url"
          onChange={[Function]}
          required={true}
          type="url"
        />
        <div
          className="photo-container"
        >
          <img
            alt="Profile"
            src=""
          />
        </div>
      </section>
    </form>
  </div>,
]
`;
